app/controllers/ticket_comments_controller.rb

```diff
@@ -29,4 +26,25 @@ def create
 
     render json: comment
   end
+
+  private def ticket_and_acting_stakeholder_from_params
+    ticket = Ticket.find(params.require(:ticket_id))
+    acting_stakeholder = TicketStakeholder.find(params.require(:acting_stakeholder_id))
+
+    [ticket, acting_stakeholder]
+  end
+
+  private def check_ticket_errors(ticket, acting_stakeholder, action)
```

Comment:
This is probably best done as part of the Rails filter chain. The idea is to declare a `before_action :check_ticket_errors, only: [:create]` (where the `only` part can be adjusted as needed), because the Rails controller chain automatically halts and returns early, if any `render` or `redirect` has been called during a `before_action`.
It effectively gets rid of the need to return a boolean for you.

Reply:
Also, when you want to use variables from database calls, like `ticket_and_acting_stakeholder_from_params`, you can do this:
- In your `before_action`, you store these variables into `@ticket` instance variables.
- If the filter chain halts, because some of your business logic checks failed, then Rails will just return early and the response will already go out to the client (ie React code in the browser). The `@ticket` variable is a little bit "useless", compared to a normal `ticket` local variable.
- But if the filter chain does not halt, and all pre-checks pass, then the `@ticket` declared in the `before_action` method body will also be available in the normal controller method body without any additional action needed.

Reply:
(Sidenote: In an upcoming Rubocop release, the linter will actually fail on these "force to return a dummy boolean just so that I can return early in the controller" methods, because it's not considered good practice in Rails. See some controller files in https://github.com/thewca/worldcubeassociation.org/pull/11962/files for reference)